### ISSUE TITLE
Need option to change the default SOCKS port for Orbot

### ISSUE THREAD

**Issue by SillaRizzoli** [guardianproject/ChatSecureAndroid#679]
Hello,
can you please add an option to specify the SOCKS port to use when connecting through Orbot?
This is needed because many Samsung devices already use TCP port 9050 for some Samsung service; it is simple enough to modify this setting in Orbot, there's an option in the GUI, but it is not possible to change the default in ChatSecure.

Thanks,
Silla Rizzoli

**Comment by n8fr8** [comment]
Yes, we will expose the proxy settings value. Also, we are adding a way for apps to query Orbot to ask for the current port values.

**Comment by SillaRizzoli** [comment]
Great, thanks!
In the meantime, I've been trying to do it myself, as a kind of learning project, but I couldn't get ChatSecure to run in Eclipse; following the instructions did help some, however I had to add many libraries manually to get eclipse to let me try to run the project; when I try to run it, however, I get this error:
[2015-05-18 21:03:12 - Dex Loader] Unable to execute dex: Multiple dex files define Lcom/kenai/jbosh/AbstractAttr;
[2015-05-18 21:03:12 - ChatSecure] Conversion to Dalvik format failed: Unable to execute dex: Multiple dex files define Lcom/kenai/jbosh/AbstractAttr;

Can you help me solve this problem?

Thanks,
Silla Rizzoli

**Comment by n8fr8** [comment]
Did you follow the instructions in the README.md file?

You have to get all the submodule projects using:
 git submodule update --init

and then you need to import all of those projects into Eclipse.

**Comment by SillaRizzoli** [comment]
Yes, I did that, I just got it to work, I had to uncheck all the .jar files in the project, clean and rebuild everything.

Thanks!
